bin/node/runtime/src/lib.rs

```diff
@@ -786,6 +810,31 @@ impl_runtime_apis! {
 		fn grandpa_authorities() -> GrandpaAuthorityList {
 			Grandpa::grandpa_authorities()
 		}
+
+		fn submit_report_equivocation_extrinsic(
+			equivocation_proof: fg_primitives::EquivocationProof<
+				<Block as BlockT>::Hash,
+				NumberFor<Block>,
+			>,
+			key_owner_proof: fg_primitives::OpaqueKeyOwnershipProof,
+		) -> Option<()> {
+			let key_owner_proof = key_owner_proof.decode()?;
+
+			Grandpa::submit_report_equivocation_extrinsic(
+				equivocation_proof,
+				key_owner_proof,
+			)
+		}
+
+		fn generate_key_ownership_proof(
+			authority_id: fg_primitives::AuthorityId,
```

Comment:
We probably want to provide `set_id` as a parameter here, since @drahnr will be changing `Historical::prove` to accept `SessionIndex`. I believe we have a `SetId -> SessionIndex` lookup somewhere.

Reply:
Yeah we have a map `SetId -> Session` in the grandpa pallet which maps to the latest session in the set. I'll update this API (and keep the parameter unused for now).